x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -393,33 +403,64 @@ protected void doRun() throws Exception {
                                         TimeValue.timeValueNanos(statsCurrentTimeNanosSupplier.getAsLong() - startTimeInNanos).millis()
                                     )
                                 );
+                            }));
+                        }
+                        enqueued = true;
+                    }
+                } catch (Exception e) {
+                    logger.warn(
+                        () -> new ParameterizedMessage("unable to open file [{}], cancelling prewarming for shard [{}]", fileName, shardId),
+                        e
+                    );
+                    if (queue.size() > 0) {
+                        try {
+                            ActionRunnable<Void> warmer;
+                            while ((warmer = queue.poll(0L, TimeUnit.MILLISECONDS)) != null) {
+                                warmer.onFailure(new IOException("Shard cache prewarming cancelled"));
```

Comment:
why not pass the original exception here?

Reply:
Right - I pushed e0f8874eb43086609aa52ba6e06414b33d34ecfb